src/components/subscriptions/SubscriptionData.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { createContext, useMemo, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { fetchSubscriptionDetails, fetchSubscriptionUsers } from './data/service';
+
+import { TAB_ALL_USERS } from './constants';
+
+export const SubscriptionContext = createContext();
+export const SubscriptionConsumer = SubscriptionContext.Consumer;
+
+function transformUsers(users) {
+  const getLicensedUsers = () => users.filter(user => user.licenseStatus === 'active');
+  const getPendingUsers = () => users.filter(user => user.licenseStatus === 'assigned');
+  const getDeactivatedUsers = () => users.filter(user => user.licenseStatus === 'deactivated');
+
+  const getAllUsers = () => [
+    ...getLicensedUsers(),
+    ...getPendingUsers(),
+    ...getDeactivatedUsers(),
+  ];
+
+  return {
+    all: getAllUsers(),
+    licensed: getLicensedUsers(),
+    pending: getPendingUsers(),
+    deactivated: getDeactivatedUsers(),
+  };
```

Comment:
I thought we had planned for the API to share the number of users in each of these status buckets?

Reply:
Are you referring to the `overview` endpoint we had discussed in the API doc?

Reply:
I thought part of the users endpoint was to include total counts of each status, but it could be part of the general subscription metadata endpoint. The main thing id that I thought we were planning to rely on the backend for that info

Reply:
Yes, you are correct. I think the original intent of this PR was to be an interactive prototype that would inform the next steps of how it would need to interact with the API endpoints. The fetching the total counts of each status could have been done as follow-up work.

That said, I just pushed up a commit that makes a mock API request to fetch the total counts based on the `overview` endpoint we discussed in the doc.

I made sure it gets called on initial page load and on any search or clearing of a search. The total counts of each status now comes from a mock API endpoint.